### ISSUE TITLE
We should use the same order for column add / remove that we do for the table field ordering in viz settings

### ISSUE THREAD

**Issue by kdoh** [metabase/metabase#28619]
When trying to trim up a table to just the information you want to see I find myself having to jump between these two panels a lot. It's not particularly helpful that they are also in different orders between the two screens.

E.x. 
These should be in the same order, but they aren't.
<img width="416" alt="Screen Shot 2023-02-24 at 2 59 26 PM" src="https://user-images.githubusercontent.com/5248953/221278914-2c078861-69ca-46c9-b08e-d1548f9194c6.png">
<img width="402" alt="Screen Shot 2023-02-24 at 2 59 20 PM" src="https://user-images.githubusercontent.com/5248953/221278917-e91eb7b6-1059-4bdc-add2-73ef11ce00b3.png">

**Comment by kdoh** [comment]
Spec: The ordering of columns in the select / deselect view should match the field order from the prior view. (It feels like more of a bug in that sense).

With that here, I'm going to remove the Design Needed label.

**Comment by luizarakaki** [comment]
@kdoh 
We tested this in the scope of https://github.com/metabase/metabase/pull/34087 and the general behavior changed a bit.

I think making the add/remove columns list consistent with the viz order but inconsistent with the table column order will create another consistency issue.

Now that you can hide/show fields directly from the first layer and we have a search mechanism, I think this problem is mitigated.

If you can take a look on that branch and check, would be nice. If there are no concerns, my bias here is to close this issue.

**Comment by kdoh** [comment]
@luizarakaki Hey sorry I missed the initial ping on this. I took a look at #34087 and while there are improvements there, I don't think we've done a great job of letting people know why these lists are different. This is most apparent in the default state where I'm just looking at a single table and they do happen to match. In one case you're saying what should come back from the DB and be possible to be shown, and in the other you're saying what's going on in the visible table.

I think I'd still be likely to come file this issue like I did earlier this year without some better contextualization because that's not immediately clear. Some headings or something else might help me out there if this isn't something I'm aware of because of lack of familiarity with the concepts or just if I'm lacking some sleep and I don't realize which each does.
